### PR TITLE
Closes #2366 Allow linking images that are embedded in content.

### LINKED ISSUE #2366: Allow displaying images as linked images
## Motivation
Sometimes I just want to use an image as a link, (especially using the natural size display mode)

According to the W3C there are accessibility concerns with using images as links, however there is a definite allowable use-case with the[ functional image sub-category](https://www.w3.org/WAI/tutorials/images/functional/#logo-image-within-link-text). 

### Is your feature request related to a problem? Please describe.
I can't create sponsor logos that link to my sponsors.

## Proposed Resolution

Add an option to CKEditor for wrapping non interactable media with a link

Possible solution https://www.drupal.org/project/drupal/issues/3075527

### Describe the solution you'd like

Use built-in ckeditor link button to wrap media(non-interactable) that I embed in a link.
Not good for iframes, embeds, content, blocks etc, but should be allowed for images.

### Describe alternatives you've considered
Alternatively, we could just allow embedding images styles OR media display mode when choosing how to display images, bypassing this whole problem.

### Roles and Permissions considerations
A clear and concise description of how each of the following roles would be impacted by this change:
- Anonymous user (view embedded images)
- Authenticated user (view embedded images)
- Content editor (embed images by style rather than display mode)
- Content administrator (embed images by style rather than display mode)
- Administrator (Not sure how this affects images that are replaced after initial upload.)

## Additional context
Fairly good guide on how to use linked images https://sc.edu/about/offices_and_divisions/digital-accessibility/guides_tutorials/alternative_text/linked-images/index.php

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Rearranging filters seems to do the trick as mentioned here: https://www.drupal.org/project/drupal/issues/3075527#comment-14292415

## Related issues
Closes #2366

## How to test
Embed media into a ckeditor field using the az_standard text formatter, and select natural size (or any other display) and then select the embeded media in the editor, then use the link button to add a link.

Save
Click image
Go to the link you used.

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_media/config/install/filter.format.az_standard.yml [hunk 1]
```diff
@@ -20,35 +21,35 @@ filters:
...
...class> <ol start type class> <li class> <dl> <dt> <dd> <h1 name id class> <h2 name id class> <h3 name id class> <h4 name id class> <h5 name id class> <h6 name id class> <sup> <sub> <table class> <caption> <tbody class> <thead class> <tfoot class> <th scope colspan rowspan class> <td colspan rowspan class> <tr colspan rowspan class> <hr class> <a href target name id data-entity-substitution data-entity-type data-entity-uuid title class> <span name id class> <img src alt data-entity-type data-entity-uuid> <p class> <footer class> <code> <pre> <drupal-media data-view-mode data-align data-entity-type data-entity-uuid alt title data-entity-type data-entity-uuid data-view-mode data-align data-caption alt title> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button alt title data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button data-langcode alt title data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button data-langcode alt title>'
```
Comment (trackleft) [inline modules/custom/az_media/config/install/filter.format.az_standard.yml:32]:
```suggestion
allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class=""> <br> <div class="" name="" id="" class=* data-*> <ul type class=""> <ol start type class=""> <li class=""> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table class=""> <caption> <tbody class=""> <thead class=""> <tfoot class=""> <th class="" scope="" colspan="" rowspan=""> <td class="" colspan="" rowspan=""> <tr class="" colspan="" rowspan=""> <hr class=""> <a href class="" target="" name="" id="" data-entity-substitution data-entity-type data-entity-uuid title> <span class="" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <p class=""> <footer class=""> <code> <pre>'
```

File: modules/custom/az_media/config/install/filter.format.az_standard.yml [hunk 2]
```diff
@@ -20,35 +20,35 @@ filters:
...
+allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class=""> <br> <div class="" name="" id="" class=* data-*> <ul type class=""> <ol start type class=""> <li class=""> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table class=""> <caption> <tbody class=""> <thead class=""> <tfoot class=""> <th class="" scope="" colspan="" rowspan=""> <td class="" colspan="" rowspan=""> <tr class="" colspan="" rowspan=""> <hr class=""> <a href class="" target="" name="" id="" data-entity-substitution data-entity-type data-entity-uuid title> <span class="" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <p class=""> <footer class=""> <code> <pre>'
```
Comment (trackleft) [inline modules/custom/az_media/config/install/filter.format.az_standard.yml:31]:
```suggestion
    allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class=""> <br> <div class="" name="" id="" class=* data-*> <ul type class=""> <ol start type class=""> <li class=""> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table class=""> <caption> <tbody class=""> <thead class=""> <tfoot class=""> <th class="" scope="" colspan="" rowspan=""> <td class="" colspan="" rowspan=""> <tr class="" colspan="" rowspan=""> <hr class=""> <a href class="" target="" name="" id="" data-entity-substitution data-entity-type data-entity-uuid title> <span class="" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <p class=""> <footer class=""> <code> <pre>'
```

File: modules/custom/az_media/config/install/filter.format.az_standard.yml [hunk 3]
```diff
@@ -20,35 +20,35 @@ filters:
...
+    allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class=""> <br> <div class="" name="" id="" class=* data-*> <ul type class=""> <ol start type class=""> <li class=""> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table class=""> <caption> <tbody class=""> <thead class=""> <tfoot class=""> <th class="" scope="" colspan="" rowspan=""> <td class="" colspan="" rowspan=""> <tr class="" colspan="" rowspan=""> <hr class=""> <a href class="" target="" name="" id="" data-entity-substitution data-entity-type data-entity-uuid title> <span class="" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <p class=""> <footer class=""> <code> <pre>'
```
Comment (trackleft) [inline modules/custom/az_media/config/install/filter.format.az_standard.yml:31]:
```suggestion
      allowed_html: '<em> <strong> <s> <abbr> <address> <cite> <blockquote class=""> <br> <div class="" name="" id="" class=* data-*> <ul type class=""> <ol start type class=""> <li class=""> <dl> <dt> <dd> <h1 class="" name="" id=""> <h2 class="" name="" id=""> <h3 class="" name="" id=""> <h4 class="" name="" id=""> <h5 class="" name="" id=""> <h6 class="" name="" id=""> <sup> <sub> <table class=""> <caption> <tbody class=""> <thead class=""> <tfoot class=""> <th class="" scope="" colspan="" rowspan=""> <td class="" colspan="" rowspan=""> <tr class="" colspan="" rowspan=""> <hr class=""> <a href class="" target="" name="" id="" data-entity-substitution data-entity-type data-entity-uuid title> <span class="" name="" id=""> <img src alt data-entity-type data-entity-uuid> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align="" data-caption data-embed-button alt title> <drupal-media data-view-mode="" data-align="" data-entity-type data-entity-uuid alt title> <p class=""> <footer class=""> <code> <pre>'
```

File: modules/custom/az_media/config/install/filter.format.az_standard.yml [hunk 4]
```diff
@@ -85,12 +85,48 @@ filters:
...
+    provider: filter
+    status: false
+    weight: -35
+    settings: {  }
+  filter_caption:
+    id: filter_caption
+    provider: filter
+    status: false
+    weight: -36
+    settings: {  }
```
Comment (trackleft) [inline modules/custom/az_media/config/install/filter.format.az_standard.yml:132]:
```suggestion
```

**Review by camikazegreen — APPROVED** [review]
Wasn't obvious to me how to actually select an image in the CK Editor, but once I figured that out, it worked as expected.

**Review by bberndt-uaz — APPROVED** [review]
When the image alignment is set to "Left" or "Right", only the image is linked. If the image alignment is set to "None" or "Center", the entire block is linked instead of just the image.

image

However, this is an existing issue with how images are embedded so I think it doesn't need to block this PR.

I think we do need to add documentation about the proper use cases for linked images to the Quickstart site.

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**